src/code_8002774.c

```diff
@@ -1,6 +1,101 @@
 #include "global.h"
 #include "code_8002774.h"
 #include "constants/direction.h"
+#include "structs/str_position.h"
+#include "other_random.h"
+
+typedef struct Vector
```

Comment:
There's already the `PixelPos` struct which has the same fields. That can be merged with this new struct.